server.js

```diff
@@ -1,77 +1,106 @@
 const Discord = require('discord.js');
-const client = new Discord.Client();
-const token = process.env.DISCORD_TOKEN;
 const redis = require('redis');
 const SHA256 = require('crypto-js/sha256');
-const moment = require('moment');
-const express = require('express');
+const moment = require('moment-timezone');
+
+const PATH_TO_STATIC_FOLDER = `${__dirname}/static/img/`;
+const { DISCORD_TOKEN = null, REDIS_URL = 'redis://redis:6379' } = process.env;
+const client = new Discord.Client();
+const token = DISCORD_TOKEN;
+const TWELVE_HOURS = 43200;
 
 const IS_URL_REGEX = /https?:\/\/(www\.)?[-a-zA-Z0-9@:%._\+~#=]{2,256}\.[a-z]{2,6}\b([-a-zA-Z0-9@:%_\+.~#?&\/\/=]*)/;
 const IS_AH_REGEX = /^\bah\b$/i;
-const IS_NESTCEPAS_REGEX = /^n'?estcepas$/ig;
+const IS_PROJET_REGEX = /^\bprojet\b$/i;
+const IS_NESTCEPAS_REGEX = /^n'?estcepas$/gi;
+const IS_GET_VANCOUVER_TIME_REGEX = /!vanc/;
+const IS_GET_PARIS_TIME_REGEX = /!par/;
 
 /*
  * SETUP SERVER
  */
-let redisClientoptions = {};
-if (process.env.REDIS_URL) {
-    redisClientoptions = { url: process.env.REDIS_URL };
-}
-
+const redisClientoptions = REDIS_URL ? { url: REDIS_URL } : {};
 const redisClient = redis.createClient(redisClientoptions);
 
 client.on('ready', () => {
-    console.log('DISCORD OK');
+  console.log('DISCORD OK');
 });
 
 redisClient.on('connect', () => {
-    console.log('REDIS OK');
+  console.log('REDIS OK');
 });
 
 /*
  * REAL MAGIC HAPPENS
  */
+client.on('message', (message) => {
+  if (message.content.match(IS_URL_REGEX)) {
+    const hashedUrl = SHA256(message.content).toString();
 
-client.on('message', message => {
-    if (message.content.match(IS_URL_REGEX)) {
-        const hashedUrl = SHA256(message.content).toString();
-
-        redisClient.get(hashedUrl, (err, reply) => {
-            if (!reply) {
-                let data = {
-                    'user': message.author.username,
-                    'channel': message.channel.name,
-                    'date': message.createdAt
-                };
-                redisClient.set(hashedUrl, JSON.stringify(data), 'EX', 43200);
-            } else if (message.deletable) {
-                let data = JSON.parse(reply);
-                let date = moment(new Date(data.date));
-                let additionalMessage = data.user + ' posted this in #' + data.channel + ' on ' + date.fromNow() || '';
-                message.channel.send(additionalMessage, {file: __dirname+'/static/img/no_repost.jpg'});
-                message.delete();
-            }
+    redisClient.get(hashedUrl, (err, reply) => {
+      if (!reply) {
+        // link is fresh or at least 12 hours old
+        // save the hash and links meta in redis for 12 hours
+        const data = {
+          user: message.author.username,
+          channel: message.channel.name,
+          date: message.createdAt,
+        };
+        redisClient.set(hashedUrl, JSON.stringify(data), 'EX', TWELVE_HOURS);
+      } else if (message.deletable) {
```

Comment:
Break if / else if as just two if because it's not related, plus it's ugly.

Reply:
if you break this into 2 separate if, you will delete the message everytime someone is posting a URL.

Reply:
But i'll agree with you, this parts need some "love" 😄